menoh/tensorrt/Inference.cpp

```diff
@@ -89,15 +96,143 @@ namespace menoh_impl {
                        totalTime / TIMING_ITERATIONS);
             }
         } gProfiler;
-#endif
+#endif // MENOH_ENABLE_TENSORRT_PROFILER
         static Logger gLogger;
 
+#ifdef MENOH_ENABLE_TENSORRT_MODEL_CACHING
+        std::string Inference::calc_model_hash(
+          std::unordered_map<std::string, array> const& input_table,
+          std::unordered_map<std::string, array> const& output_table,
+          menoh_impl::model_data const& model_data, config const& config) {
+            hasher hasher;
+            auto add_variable_table =
+              [](menoh_impl::hasher& h,
+                 std::unordered_map<std::string, array> const& table) {
+                  std::vector<std::pair<std::string, array>> variables(
+                    table.begin(), table.end());
+                  std::sort(variables.begin(), variables.end(),
+                            [](auto const& a, auto const& b) {
+                                return a.first < b.first;
+                            });
+                  for(auto const& p : variables) {
+                      add_container(h, p.first);
+                      h.add(static_cast<const std::uint8_t*>(
+                              static_cast<const void*>(p.second.dims().data())),
+                            p.second.dims().size());
```

Comment:
It seems that `h.add` process only part of the data, because `h.add` takes size in bytes but `p.second.dims().size()` returns number of elements.

Reply:
Right. Thank you. I was not aware of it at all.